### ISSUE TITLE
B9PartSwitch Bug

### ISSUE THREAD

**Issue by harryofbath** [CobaltWolf/Bluedog-Design-Bureau#1579]
[Part bluedog.DeltaIV.DCSS.5m] [ModuleB9PartSwitch 'textureSwitchPaint'] Duplicate subtype names detected: Blue, White, White (Alt), White (Topaz) 

Not sure what other info is necessary, except running on Linux with CKAN. v1.14.0

**Comment by Rodg88** [comment]
Probably need a copy of your ksp.log file to see what's causing it. Duplicate subtypes like this are normally installation issues, but if its only the DCSS it may be an error in another mod's patching.

**Comment by ThomasWarChild** [comment]
I'm having an issue with the B9partswitch as well (7-27-2025).  

https://imgur.com/a/45cjYgB

[https://imgur.com/a/45cjYgB](url)

**Comment by zorg2044** [comment]
this is either a mod conflict or an incorrect installation. you will need to post your logs in order to diagnose anything

**Comment by ThomasWarChild** [comment]
I uninstalled B9Partswitch several times.  This time I'll reinstall BDB and BDB Extras.

Here are my log files:  KSP Log https://drive.google.com/file/d/1uFJ_EtCzECjk-DQ-xeBlts-rLTN__eAf/view?usp=drive_link
                                             [https://drive.google.com/file/d/1uFJ_EtCzECjk-DQ-xeBlts-rLTN__eAf/view?usp=drive_lin](url)


                                    Mod list: https://drive.google.com/file/d/1hUdhJL5RyvHYvrfD1MhI7CJmRxAmnkpL/view?usp=drive_link
                                                            [https://drive.google.com/file/d/1hUdhJL5RyvHYvrfD1MhI7CJmRxAmnkpL/view?usp=drive_link](url)

**Comment by zorg2044** [comment]
> I uninstalled B9Partswitch several times. This time I'll reinstall BDB and BDB Extras.
> 
> Here are my log files: KSP Log https://drive.google.com/file/d/1uFJ_EtCzECjk-DQ-xeBlts-rLTN__eAf/view?usp=drive_link [https://drive.google.com/file/d/1uFJ_EtCzECjk-DQ-xeBlts-rLTN__eAf/view?usp=drive_lin](url)
> 
> ```
>                                 Mod list: https://drive.google.com/file/d/1hUdhJL5RyvHYvrfD1MhI7CJmRxAmnkpL/view?usp=drive_link
>                                                         [https://drive.google.com/file/d/1hUdhJL5RyvHYvrfD1MhI7CJmRxAmnkpL/view?usp=drive_link](url)
> ```

B9 partswitch itself is almost never the issue so reinstalling that wont help. Its the victim rather than the culprit. The b9 error is basically reporting a problem almost always caused by another mod or a bad installation. 

If reinstalling BDB hasn't helped could you please upload your modulemanager.configcache (found directly inside gamedata), modulemanager.log and MMPatch.log (found in Kerbal Space Program/Logs/ModuleManager).

also which specific BDB extras were installed?

**Comment by ThomasWarChild** [comment]
@zorg2044 Thank you again for your assistance.  I purged and then reinstalled the following two mods using CKAN:
BDB Version 1.14.0
BDB Extras - Community category kit - author: Rodger  V1.14.0

Same warning message appears.
Here are the links to the 3 files you asked for:

Modulemanager.configcache   https://drive.google.com/file/d/10D52RKwHvpTn149Jj1nXp9eUJNhU5kcz/view?usp=drive_link

Modulemanager.log  https://drive.google.com/file/d/1kk2e5TdedJla3S8o0BlJfBzBobHxWOjN/view?usp=drive_link
MMPatch.log  https://drive.google.com/file/d/10TJ9ZjJ_KjpGKOzZHGPFd2pSofJGV8Xv/view?usp=drive_link

Thank you all again!

**Comment by Rodg88** [comment]
I can't really see any issues in configcache, the parts showing errors seem to be formed correctly. Unless between your upload of KSP.log and the configcache the errors have changed.
It just be running out of memory due to high amount of mods, or it's a weird interaction with some 3rd mod getting in the way of B9PS's functions during load.

If you just skip past the error, those parts *could* continue to function normally

**Comment by ThomasWarChild** [comment]
@Rodg88 
Well I just reloaded BDB and BDB Extras Community Catagory Kit today and I'm still getting the errors.
Everything was working without errors on 4/30/2025.
Since then I haven't installed anything new beyond Firefly.  

Anytime I install a new mod... I first make a mod save list... I then label all the mods in CKAN called working without errors.
This way if I install something that doesn't work... I know which one it is and can remove it.
I just looked over my mod list and RO Utils showed up not labeled which would indicate that I didn't choose to install it... but rather a dependency.  Derivative Textures, Blackheart612 Procedural parts/fairings require that, so I'm at a loss as to what's causing the errors.  My PC:

Processor	AMD Ryzen 7 5800X 8-Core Processor                3.80 GHz
Installed RAM	128 GB
Storage	1.82 TB SSD Samsung SSD 870 EVO 2TB, 1.82 TB HDD ST2000DM006-2DM164, 932 GB SSD Seagate FireCuda 520 SSD ZP1000GM30002
Graphics Card	NVIDIA GeForce RTX 3080 (10 GB)
System Type	64-bit operating system, x64-based processor WIN10

Also... since posting, I have been able to ignore the error (close it) and keep playing - haven't noticed much of an issue.

Just ran HWinfo while the game is loading up... I'm using less than 32% memory with a memory load of less than 30%

**Comment by Rodg88** [comment]
If you uninstall Firefly does the issue go away? Might need to report to them instead, if installing it reliably causes the issues. It's a more complex mod than meets the eye, I could certainly see unforeseen issues being a possibility.

**Comment by ThomasWarChild** [comment]
@Rodg88   Nope... Just uninstalled Firefly and I'm still getting the errors.  This is really puzzling.

**Comment by Spriteanon** [comment]
Hey, I was looking into a similar issue (in fact, one seemingly identical to the one experienced by @ThomasWarChild and I believe I figured out what's causing it for us but why it might not show up for devs, despite this being an entire isolated issue to BDB and its prerequisites:
It's to do with the recent adoption and continuation of B9PartSwitcher.
After seeing that Thomas posted their issue recently, and having messed around with my mod list, I started going step by step and realized that B9PartSwitch had two recent updates as well. (More recent than this thread, so I believe ThomasWarChild and my issue are separate from the original issue listed here.)
As a test, I reinstalled the last version of B9PartSwitch from before the recent adoption, and the error at startup disappeared.

I haven't checked through the code completely, so I'm not entirely sure why the new version of B9PartSwitcher appears to complain about parts of BDB, but it definitely appears that it only started doing it since the most recent additions.
Sadly, the new repo for B9PartSwitcher has no public Issues tab so I can't easily report this possible glitch there, but this might also be something that needs to be addressed here.

@Rodg88 I'm kind of going out on a limb with my assumptions here, but I wouldn't be surprised if you and other devs don't manage your installations with CKAN, in which case the new version of B9PartSwitcher might have flown under the radar.

**Comment by ThomasWarChild** [comment]
@Spriteanon Thank you for your input... I hope that it helps in the end run.

**Comment by zorg2044** [comment]
Resolved in https://github.com/CobaltWolf/Bluedog-Design-Bureau/issues/1602#issuecomment-3256907876

**Comment by JonnyOThan** [comment]
@zorg2044 This shouldn't have been closed - as far as we can tell, the new versions of B9PS are catching errors in B9PS configs that the old versions weren't.  These are real bugs in BDB.
